src/components/budget-expiry-notification/index.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useMemo } from 'react';
+import {
+  ActionRow,
+  Alert, AlertModal,
+  Button, Hyperlink,
+  useToggle,
+} from '@openedx/paragon';
+
+import { sendEnterpriseTrackEvent } from '@edx/frontend-enterprise-utils';
+import { getConfig } from '@edx/frontend-platform/config';
+import useExpiry from './data/hooks/useExpiry';
+import { useEnterpriseCustomer, useHasAvailableSubsidiesOrRequests } from '../app/data';
+import { EVENT_NAMES } from './data/constants';
+import { getContactEmail } from '../../utils/common';
+
+const BudgetExpiryNotification = () => {
+  const [modalIsOpen, modalOpen, modalClose] = useToggle(false);
+  const [alertIsOpen, alertOpen, alertClose] = useToggle(false);
+  const { data: enterpriseCustomer } = useEnterpriseCustomer();
+  const { learnerCreditSummaryCardData } = useHasAvailableSubsidiesOrRequests();
+  const budget = useMemo(() => ({ end: learnerCreditSummaryCardData?.expirationDate }), [learnerCreditSummaryCardData]);
+  const {
+    alert, modal, dismissModal, dismissAlert,
+  } = useExpiry(
+    enterpriseCustomer.uuid,
+    budget,
+    modalOpen,
+    modalClose,
+    alertOpen,
+    alertClose,
+  );
+  const { LEARNER_SUPPORT_URL } = getConfig();
+  const contactEmail = getContactEmail(enterpriseCustomer);
+
+  const AlertTitle = alert?.title;
+  const ModalTitle = modal?.title;
+  const AlertMessage = alert?.message;
+  const ModalMessage = modal?.message;
+
+  const trackEventMetadata = useMemo(() => {
+    if (modal === null && alert === null) { return {}; }
+    return {
+      modal,
+      alert,
+    };
+  }, [modal, alert]);
+
+  return (
+    <>
+      {alert && (
+        <Alert
+          variant={alert.variant}
+          show={alertIsOpen}
+          actions={[
+            <Button
+              as={Hyperlink}
+              destination={LEARNER_SUPPORT_URL}
+              onClick={() => sendEnterpriseTrackEvent(
+                EVENT_NAMES.clickedToContactSupportAlert,
+                trackEventMetadata,
+              )}
```

Comment:
[nit] The first parameter of `sendEnterpriseTrackEvent` requires a `enterpriseUUID`. That could probably be brought in relatively quick by importing `useEnterpriseCustomer` in the component.